src/pages/MyProfile.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+const MyProfile = () => {
+  const { rockets } = useSelector((state) => state.rockets);
+  const { missions } = useSelector((state) => state.missions);
+  const rocketList = Object.keys(rockets);
+  const missionList = Object.keys(missions);
+  const reservedRockets = [];
+  const joinedMissions = [];
+
+  if (rocketList.length) {
+    rocketList.forEach((item) => {
+      if (rockets[item].reserved) {
+        reservedRockets.push({
+          id: reservedRockets.length + 1,
+          name: rockets[item].name,
+        });
+      }
+    });
+  }
+
+  if (missionList.length) {
+    missionList.forEach((item) => {
+      if (missions[item].joined) {
+        joinedMissions.push({
+          id: joinedMissions.length + 1,
+          name: missions[item].name,
+        });
+      }
+    });
+
+    return (
+      <div className="mycontainer">
+        <div className="mycontainer-section">
+          <h2 className="mycontainer-section__title">
+            My Missions
+          </h2>
+          <ul className="mycontainer-section__box">
+            {
+              joinedMissions.length ? joinedMissions.map((item) => (
+                <li key={item.id} className="element-name">{item.name}</li>
+              )) : <li className="element-name">No Missions Ongoing</li>
+            }
+          </ul>
+        </div>
+        <div className="mycontainer-section">
+          <h2 className="mycontainer-section__title">
+            My Rockets
+          </h2>
+          <ul className="mycontainer-section__box">
+            {
+              reservedRockets.length ? reservedRockets.map((item) => (
+                <li key={item.id} className="element-name">{item.name}</li>
+              )) : <li className="element-name">No rockets selected</li>
+            }
+          </ul>
```

Comment:
- Hey, great job with the project, you've done well so far, however, there's a part of the project requirement that has not been implemented yet. please kindly ensure you've use the `filter` method to render your rocket and mission to meet the project requirement below::

![filter](https://user-images.githubusercontent.com/59397159/180508205-b513825b-ffe0-4e71-a217-2f9395e62083.png)

Reply:
Hi @zieeco! 😁 
Thank you very much for the review! The refactorization using the `filter()` to render rockets and missions in the `My Profile` page is now working.